update spec for the 0.5.0 release

surajssd opened the pull request:
update spec to new release
change the release tag to not include the git commit
add support for bash auto completion

surajssd committed:
update spec for the 0.5.0 release

update spec to new release

surajssd commented on kompose.spec:
```diff
@@ -967,6 +973,18 @@ export LDFLAGS=%{ldflags}
+  %{_bindir}/kompose completion zsh > %{_datadir}/zsh/site-functions/kompose;
+else
+  %{_bindir}/kompose completion bash > %{_datadir}/bash-completion/completions/kompose;
+fi
```
@dustymabe @cdrage would like your input on this section!

dustymabe replied:
so my thoughts are we should create these files earlier in the process and include them in the rpm. so basically we compile kompose and the use the binary to create these files. then include them in %files. Also, lets create both zsh and bash completion files to be delivered by the rpm.

dustymabe replied:
so in `%build`, right after `%gobuild` then run `bin/kompose completion bash > file` and then in `%install` do something like `install` file into appropriate location. Under `%files` add them just like we do `%{_bindir}/kompose`

dustymabe replied:
also, add the bash completion in a 2nd commit please.

dustymabe commented on kompose.spec:
```diff
@@ -967,6 +973,18 @@ export LDFLAGS=%{ldflags}
 %{_bindir}/kompose
 
+%post
+if [ -z ${BASH+x} ];
```
hmm. what does `${BASH+x}` give you? looks like `x` if $BASH is defined and `` if not. What is the benefit of doing this and not `-z "$BASH"` ?

surajssd replied:
read about it here http://stackoverflow.com/a/13864829/3848679

dustymabe replied:
yeah but that is an answer to the question `How do I know if a variable is set in Bash?`. We don't care if the variable is set or not, just that it is not empty :). Good to know how to do that, though.

dustymabe commented on kompose.spec:
```diff
@@ -52,8 +52,8 @@
 
 Name:           kompose
-Version:        0.4.0
-Release:        0.1.git%{shortcommit}%{?dist}
```
can you remove the shortcommit in a 2nd commit? I want this to be a clear change we decided to do and not something that could be confused as a mistake.

surajssd replied:
done! working on the auto complete part!

surajssd committed:
remove git commit in release tag

To make it consistent with packaging in general
removing the git commit in release name.

surajssd committed:
add bash/zsh auto completion in rpm spec

surajssd commented:
@dustymabe done

surajssd commented:
I tested it I have to start bash anew for auto complete to work! Needs to solve that.
